pkg/cable/libreswan/libreswan.go

```diff
@@ -359,6 +427,166 @@ func whack(args ...string) error {
+		stanza += "\n"
+	}
+
+	_, err = f.WriteString(stanza)
+
+	return errors.Wrap(err, "error writing to file")
+}
+
+func removeConnectionStanza(confPath, connName string) error {
+	data, err := os.ReadFile(confPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+
+		return errors.Wrap(err, "error reading file")
+	}
+
+	lines := strings.Split(string(data), "\n")
+	var out []string
+	inStanza := false
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "conn ") && strings.TrimSpace(line) == ("conn "+connName) {
+			inStanza = true
+			continue
+		}
+
+		if inStanza && strings.HasPrefix(line, "conn ") && strings.TrimSpace(line) != ("conn "+connName) {
+			inStanza = false
+		}
+
+		if !inStanza {
+			out = append(out, line)
+		}
+	}
+
+	for len(out) > 0 && out[len(out)-1] == "" {
+		out = out[:len(out)-1]
+	}
+
+	if len(out) == 0 {
+		return errors.Wrap(os.Remove(confPath), "error removing file")
+	}
+
+	return errors.Wrap(os.WriteFile(confPath, []byte(strings.Join(out, "\n")+"\n"), 0o600), "error writing file")
+}
+
+func (i *libreswan) connectToEndpointCertMode(endpointInfo *natdiscovery.NATEndpointInfo) (string, error) {
+	logger.Info("Certificate mode: skipping pluto start and whack; assuming pluto is managed externally and config/NSS DB are updated.")
+
+	confPath := submarinerConfPath
+	endpoint := &endpointInfo.Endpoint
+	leftID := "submariner-client-" + i.localEndpoint.ClusterID
+	left := i.localEndpoint.GetPrivateIP(endpointInfo.UseFamily)
+	right := endpointInfo.UseIP
+	leftSubnets := i.localEndpoint.Subnets
+	rightSubnets := endpoint.Spec.Subnets
+
+	for lsi, leftSubnet := range leftSubnets {
+		for rsi, rightSubnet := range rightSubnets {
+			connName := toConnectionName(endpoint.Spec.CableName, endpointInfo.UseFamily, lsi, rsi)
+
+			encapsulationLine := ""
+			if endpointInfo.UseNAT || i.forceUDPEncapsulation {
+				encapsulationLine = "    encapsulation=yes\n"
+			}
+
+			conf := fmt.Sprintf(`conn %s
+    left=%s
+    leftid=%%fromcert
+    leftcert=%s
+    leftrsasigkey=%%cert
+    leftsubnet=%s
+    leftmodecfgclient=false
+    right=%s
+    rightid=%%fromcert
+    rightsubnet=%s
+%s    auto=add
+    ikev2=insist
+    authby=rsasig
+    type=tunnel`,
+				connName,
+				left,
+				leftID,
+				leftSubnet,
+				right,
+				rightSubnet,
+				encapsulationLine,
+			)
+			if err := appendConnectionStanza(confPath, conf, connName); err != nil {
+				logger.Errorf(err, "Failed to append connection stanza to %s", confPath)
+				return "", err
+			}
+
+			logger.Infof("Appended Libreswan connection config for %s to %s", connName, confPath)
+
+			cmd := exec.Command("ipsec", "auto", "--add", connName)
+
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				logger.Errorf(err, "Failed to add connection with ipsec auto --add: %s", string(output))
+				return "", errors.Wrap(err, "failed to add connection")
+			} else {
+				logger.Infof("Added connection with ipsec auto --add: %s", string(output))
+			}
```

Comment:
No need for else here.


```suggestion
			} 

             logger.Infof("Added connection with ipsec auto --add: %s", string(output))
```